### DISCUSSION TITLE
Option for Matrix field blocks to open in slideout

### DISCUSSION THREAD

**Discussion by thupsi** [craftcms/cms#10196] [Ideas] [answered]
I'm aware of (and excited about) the upcoming Matrix UI modes. I have also read some discussions about complex Matrix blocks becoming awkward in terms of UI for authors. 

My suggestion is to provide an **option/ability for Matrix blocks to open in a slideout** instead of having the collapse/expand behaviour. So, essentially, in this scenario the list of matrix blocks in an entry would look like a structure (so no need to introduce a new UI paradigm), and by double clicking on each block a slideout with all the fields would open.

To make things clear, I'm not suggesting for this to become the default behaviour, just an option per matrix field, or perhaps per matrix block (I _think_ the former would be more sensible?).

**Benefits**:
- Uncluttered UI
- Progressive disclosure of information
- Easier to scan a long list of Matrix blocks
- Easier to visually digest the hiearchy of nested Matrix blocks (when that feature lands)
- Easier to get an overview of an entry that mixes Matrix with other custom fields

I don't know if this has already been considered in relation to the upcoming Matrix UI improvements, but if it hasn't, I think it could be an easy win for usability in certain cases.

**Comment by brandonkelly** [comment, marked as answer]
Definitely something we are considering ;)
